mlflow/models/evaluation/default_evaluator.py

```diff
@@ -94,40 +94,25 @@ def _extract_raw_model(model):
 
 
 def _extract_predict_fn(model, raw_model):
-    og_predict_fn = model.predict
-    og_predict_proba_fn = None
+    predict_fn = model.predict
+    predict_proba_fn = None
 
     if raw_model is not None:
-        og_predict_fn = raw_model.predict
-        og_predict_proba_fn = getattr(raw_model, "predict_proba", None)
+        predict_fn = raw_model.predict
+        predict_proba_fn = getattr(raw_model, "predict_proba", None)
 
         try:
             import xgboost
 
             if isinstance(raw_model, xgboost.XGBModel):
                 # Because shap evaluation will pass evaluation data in ndarray format
                 # (without feature names), if set validate_features=True it will raise error.
-                og_predict_fn = partial(og_predict_fn, validate_features=False)
-                if og_predict_proba_fn is not None:
-                    og_predict_proba_fn = partial(og_predict_proba_fn, validate_features=False)
+                predict_fn = partial(predict_fn, validate_features=False)
+                if predict_proba_fn is not None:
+                    predict_proba_fn = partial(predict_proba_fn, validate_features=False)
         except ImportError:
             pass
 
-    # Wrap the original predict_fn / predict_proba_fn, add `input_df.copy()`
-    # preprocessing because some model might modify the input dataframe
-    # (e.g., appending new columns, or modify column values).
-
-    def predict_fn(input_df):
-        return og_predict_fn(input_df.copy())
-
-    if og_predict_proba_fn is not None:
-
-        def predict_proba_fn(input_df):
-            return og_predict_proba_fn(input_df.copy())
-
-    else:
-        predict_proba_fn = og_predict_proba_fn
```

Comment:
Remove one-of `pd.DataFrame.copy()` statements in favor of introducing an intermediate `_MutationGuardedData` class that *only* provides access to data by way of copying (see below).